### ISSUE TITLE
Regression when generating typescript client for unknown generic collection

### ISSUE THREAD

**Issue by psjmb** [anymaniax/orval#1286]
## What are the steps to reproduce this issue?
When generating a TypeScript client with an entity that utilizes a mapped type, I encounter a TypeScript error. This issue did not occur in previous versions.

## Steps to Reproduce

1. Generate a TypeScript client.
2. Use an entity that employs a mapped type ie:
```
/components/schemas/ProblemDetails:
{
    "type": "object",
    "properties": {
        "type": {
            "type": "string",
            "nullable": true
        },
        "title": {
            "type": "string",
            "nullable": true
        },
        "status": {
            "type": "integer",
            "format": "int32",
            "nullable": true
        },
        "detail": {
            "type": "string",
            "nullable": true
        },
        "instance": {
            "type": "string",
            "nullable": true
        }
    },
    "additionalProperties": {}
}
```
my orval's config file:
```
module.exports = {
  'api-react-query': {
    input: { target: './backend-api.json' },
    output: {
      client: 'react-query',
      target: './backend-api.ts',
      mode: 'tags-split',
      prettier: true,
      override: {
        mutator: {
          path: 'axios.ts',
          name: 'ApiAxiosInstance',
          default: false
        }
      }
    }
  }
```
The generated `schema.ts` now looks like this:
```
export interface ProblemDetails {
  detail?: string | null;
  instance?: string | null;
  status?: number | null;
  title?: string | null;
  type?: string | null;
  [key: string]: {};
}
```

However, I’m encountering the following TypeScript error:
`TS2411: Property detail of type string | null | undefined is not assignable to string index type {}`

## What were you expecting to happen?
Previously, the interface was generated as follows:
```
export interface ProblemDetails {
  detail?: string | null;
  instance?: string | null;
  status?: number | null;
  title?: string | null;
  type?: string | null;
  [key: string]: unknown;
}
```

There were no TypeScript errors.

## What versions are you using?
OS: Win10
Orval: 6.25
.NET 8
Swashbuckle.AspNetCore Version 6.5.0
using build in framework class ProblemDetails

**Comment by melloware** [comment]
@soartec-lab i wasn't sure if this was related to something you already fixed?

**Comment by danicc097** [comment]
Probably related to #1207

**Comment by melloware** [comment]
@danicc097 i think you are right and this is probably alreayd fixed in upcoming 6.26

**Comment by soartec-lab** [comment]
Yes. this issue fixed by bellow:

https://github.com/anymaniax/orval/pull/1228

It is scheduled to be released in 6.26.0, so please wait a little longer 👍

**Comment by soartec-lab** [comment]
@psjmb 

Thank you for made this report. This issue has been resolved, so I will close it for now. Please let me know if you need anything else.
